metrics-exporter/src/bin/io_engine/cache/mod.rs

```diff
@@ -0,0 +1,87 @@
+mod pool;
+
+use crate::{
+    client::{grpc_client::GrpcClient, pool::Pools},
+    ExporterConfig,
+};
+
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+use std::sync::Mutex;
+use tokio::time::sleep;
+use tracing::error;
+
+static CACHE: OnceCell<Mutex<Cache>> = OnceCell::new();
+
+/// Trait to be implemented by all Resource structs stored in Cache.
+pub trait ResourceOps {
```

Comment:
Why are these public? and many more across the file

Reply:
This is not public now. Its used within module. For others i have made crate-public.